kroxylicious-kms-provider-hashicorp-vault/src/main/java/io/kroxylicious/kms/provider/hashicorp/vault/VaultEdekSerde.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.hashicorp.vault;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+
+import io.kroxylicious.kms.service.Serde;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+class VaultEdekSerde implements Serde<VaultEdek> {
+
+    @Override
+    public VaultEdek deserialize(@NonNull ByteBuffer buffer) {
+        // TODO use varints once approach is agreed.
+        short kekRefLength = Serde.getUnsignedByte(buffer);
+        if (kekRefLength < 1) {
+            throw new IllegalArgumentException("Unexpected key ref length (%d) read deserializing VaultEdek :".formatted(kekRefLength));
+        }
+        var buf = new byte[kekRefLength];
```

Comment:
* Does Vault publish a maximum allows key id length? If so we should enforce it here, especially once the length can be more than a single byte.
* `buf` and `buffer` are a little too similar imho. `kekRefBytes`?